prometheus_juju_exporter/exporter.py

```diff
@@ -1,32 +1,76 @@
 import asyncio
+from logging import getLogger
 
-from prometheus_client import CollectorRegistry, start_http_server
+from prometheus_client import CollectorRegistry, Gauge, start_http_server
 
-from prometheus_juju_exporter.collector import CollectorDaemon
+from prometheus_juju_exporter.collector import Collector
 from prometheus_juju_exporter.config import Config
-from prometheus_juju_exporter.logging import get_logger
 
 
 class ExporterDaemon:
     """Core class of the exporter daemon."""
 
-    def __init__(self, debug=False):
+    def __init__(self):
         """Create new daemon and configure runtime environment."""
         self.config = Config().get_config()
-        self.logger = get_logger(debug=debug)
+        self.logger = getLogger(__name__)
         self.logger.info("Parsed config: %s", self.config.config_dir())
         self._registry = CollectorRegistry()
-        self.collector = CollectorDaemon(self._registry, debug=debug)
+        self.metrics = {}
+        self.collector = Collector()
         self.logger.debug("Exporter initialized")
 
+    def _create_metrics_dict(self, gauge_name, gauge_desc, labels=[]):
```

Comment:
I missed this in my last review, but[ mutable objects are not suitable for parameter defaults](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).